extension/src/Logging/OutputLogger.ts

```diff
@@ -0,0 +1,40 @@
+import { OutputChannel, window } from "vscode";
+import { ILogger } from "./ILogger";
+import { appendTimestamp } from "./LogHelper";
+
+export class OutputLogger implements ILogger {
+  private static instance: OutputLogger;
+  private channel: OutputChannel;
+  private static readonly channelName: string = "NAB AL Tools";
+
+  static getInstance(): OutputLogger {
+    if (!this.instance) {
+      this.instance = new OutputLogger();
+    }
+    return this.instance;
+  }
+
+  private constructor() {
+    this.channel = window.createOutputChannel(OutputLogger.channelName);
+  }
+  log(message?: string, ...optionalParams: string[]): void {
+    if (message) {
+      this.channel.appendLine(appendTimestamp(message));
+      for (const line of optionalParams) {
+        this.channel.appendLine(line);
+      }
+    } else {
+      this.channel.appendLine("");
```

Comment:
how come we append empty string?

Reply:
Just to force an empty line.

Reply:
This is for the case where we're actually want an empty line in the output? I was thinking this might inadvertently make a null/undefined value less visible without timestamp. But that's probably a non issue right?

Reply:
Yes, Only if we want an empty line. Not used anywhere right now... We'll change later if needed.